File: lib/domain/cultivation_plan/dtos/public_plan_save_header_snapshot.rb
# frozen_string_literal: true

module Domain
  module CultivationPlan
    module Dtos
      # PublicPlanSaveReadGateway の find_header 戻り値。
      class PublicPlanSaveHeaderSnapshot
        attr_reader :plan_id, :farm_id

        def initialize(plan_id:, farm_id:)
          @plan_id = plan_id.to_i
          @farm_id = farm_id&.to_i
          freeze
        end
      end
    end
  end
end

File: crates/agrr-domain/src/cultivation_plan/dtos/public_plan_save_header_snapshot.rs
//! Ruby: `Domain::CultivationPlan::Dtos::PublicPlanSaveHeaderSnapshot`

/// Ruby: `Domain::CultivationPlan::Dtos::PublicPlanSaveHeaderSnapshot`
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub struct PublicPlanSaveHeaderSnapshot {
    pub plan_id: i64,
    pub farm_id: Option<i64>,
}

impl PublicPlanSaveHeaderSnapshot {
    pub fn new(plan_id: i64, farm_id: Option<i64>) -> Self {
        Self { plan_id, farm_id }
    }
}
